alternator/executor.cc

```diff
@@ -1135,8 +1135,18 @@ static future<executor::request_return_type> create_table_on_shard0(tracing::tra
     }
     co_await service::prepare_new_column_family_announcement(schema_mutations, sp, *ksm, schema, ts);
     for (schema_builder& view_builder : view_builders) {
+        view_ptr view(view_builder.build());
         db::schema_tables::add_table_or_view_to_schema_mutation(
-            view_ptr(view_builder.build()), ts, true, schema_mutations);
+            view, ts, true, schema_mutations);
+        // add_table_or_view_to_schema_mutation() is a low-level function that
+        // doesn't call the callbacks that prepare_new_view_announcement()
```

Comment:
is there a reason to not use prepare_new_view_announcement here?

Reply:
Unfortunately, yes. Each of these functions makes some very specific assumptions - for example prepare_new_view_announcement works like CREATE VIEW - adding a view to an existing table. Alternator creates a keyspace, base-table and view all in one operation.